corehq/apps/ivr/models.py

```diff
@@ -140,3 +140,32 @@ def _migration_get_fields(cls):
     @classmethod
     def _migration_get_couch_model_class(cls):
         return CallLog
+
+    @classmethod
+    def by_gateway_session_id(cls, gateway_session_id):
+        result = cls.objects.filter(
+            gateway_session_id=gateway_session_id
+        ).order_by('-date')[:1]
+
+        if result:
+            return result[0]
+
+        return None
+
+    @classmethod
+    def answered_call_exists(cls, contact_doc_type, contact_id, from_timestamp, to_timestamp=None):
+        qs = cls.by_recipient(
+            contact_doc_type,
+            contact_id
+        ).filter(
+            direction=OUTGOING,
+            date__gte=from_timestamp,
+            answered=True
+        )
+
+        if to_timestamp:
+            qs = qs.filter(
+                date__lte=to_timestamp
+            )
+
+        return len(qs[:1]) > 0
```

Comment:
my thinking here was that the slicing for `qs[:1]` forces it to only fetch the first record to determine if it exists or not (`len(qs)` forces it to fetch all records).  but now that i think about it, it's probably just as good to use qs.count() here. will update